### PR TITLE
fix: revise call signature of addChunkInGroup

### PR DESCRIPTION
**Pull request by forivall**
**What kind of change does this PR introduce?**

Add a missing overload signature for addChunkInGroup's typescript declaration.

**Did you add tests for your changes?**

N/A (unless the definitions are tested somewhere)

**Does this PR introduce a breaking change?**

N/A

**What needs to be documented once your changes are merged?**

This is basically a documentation fix.

### CODE REVIEW

**Comment by alexander-akait** [comment]
Please add types using ts jsdocs, don't modify them directly

**Review by sokra — CHANGES_REQUESTED** [review]
You need to fix that in the Code here: https://github.com/forivall/webpack/blob/5c0767a4239adfd239d80fa605a88c2262c54022/lib/Compilation.js#L3555-L3562
and run `yarn special-lint-fix` to re-generate the typings

**Comment by forivall** [comment]
I dont think typescript has any way of understanding class method overloads with jsdoc ([it's kinda janky anyways](https://github.com/jsdoc/jsdoc/issues/1017)), so i just made the arguments optional and left documentation. (I took a look through the typedef generation code to see that you're fetching the jsdoc using typescript itself. I also determined that adding some kind of hack to make it work seems unreasonable)

File: types.d.ts [hunk 1]
```diff
@@ -1644,11 +1644,15 @@ declare class Compilation {
...
+	/**
+	 * If `module` is passed, `loc` and `request` must also be passed.
+	 */
 	addChunkInGroup(
 		groupOptions: string | ChunkGroupOptions,
-		module: Module,
-		loc: DependencyLocation,
-		request: string
+		module?: Module,
+		loc?: SyntheticDependencyLocation | RealDependencyLocation,
```
Comment (forivall) [inline types.d.ts:1654]: hmm, this seems to have been caused by the generation script. union normalization?

Comment (sokra) [reply types.d.ts:1654]: yes, typescript isn't the best in keeping names. An optional argument adds `undefined` as possible value. So `DependencyLocation | undefined` is normalized to `SyntheticDependencyLocation | RealDependencyLocation | undefined` and for print `loc?` the `undefined` is omitted again => `SyntheticDependencyLocation | RealDependencyLocation`. But that doesn't matter.

**Review by sokra — APPROVED** [review]

**Comment by sokra** [comment]
Thanks

**Pull request merged**